### PR TITLE
Fixes #3900: Add file size info to addon's "More Info" section

### LINKED ISSUE #3000: Add Homepage carousel
Primary components of the new Homepage carousel will include:

- Headline is the only copy element needed 
- @pwalm will experiment with designs to see if a CTA is required; would it be too minimalist or un-engaging to eschew?

[PRD](https://docs.google.com/document/d/1JLNsuRbd8m4uG-HW5j67_sAT4wb1z-pAj0YaHd_76wE/edit#heading=h.ifofy8kjhyvh) for reference

@pwalm @muffinresearch

### LINKED ISSUE #3900: Expose the Addon size in the Add-on metadata
Refiled from https://github.com/mozilla/addons/issues/535

On the old site we exposed the add-on download size. 

ublock_origin____add-ons_for_firefox

However, in the new frontend this detail is missing. It's useful for knowing the size of the download ahead of time so it would be good to re-instate this data.

### PR DESCRIPTION
**Pull request by voronianski**
Fixes #3900 

This patch adds file size information to "More Info" section on addon's page.

Before:

Screenshot 2019-06-29 at 14 37 54 

After:

Screenshot 2019-06-29 at 14 36 42

### Approach and changes

- [x] updated `AddonMoreInfo` component with `versionInfo` prop
- [x] using `getVersionInfo` reducer for this purpose similarly to `AddonVersionCard` component [which is already capable](https://github.com/mozilla/addons-frontend/blob/master/src/amo/components/AddonVersionCard/index.js#L204-L211) of displaying such data
- [x] added relevant test case into `TestAddonMoreInfo` to cover the change

### CODE REVIEW

File: tests/unit/amo/components/TestAddonMoreInfo.js [hunk 1]
```diff
@@ -184,6 +185,20 @@ describe(__filename, () => {
...
+      files: [
+        {
+          ...fakePlatformFile,
+          size: 10,
+        },
+      ],
+    });
+    const root = render({});
+
+    expect(root.find('.AddonMoreInfo-filesize').children()).toHaveText('10 B');
```
Comment (willdurand) [inline tests/unit/amo/components/TestAddonMoreInfo.js]: Could you please use the `formatFilesize` (from `core/i18n/utils`) in `toHaveText`?

```js
expect(root.find('.AddonMoreInfo-filesize').at(0)).toHaveText(formatFilesize(size));
```

In the snippet above, I also extract `size` as a variable so that you can use it in the `_loadVersions` call as well.

File: src/amo/components/AddonMoreInfo/index.js [hunk 2]
```diff
@@ -278,20 +291,37 @@ export class AddonMoreInfoBase extends React.Component<InternalProps> {
...
-      addon && addon.currentVersionId
-        ? getVersionById({ id: addon.currentVersionId, state: state.versions })
-        : null,
   };
 };
 
 const AddonMoreInfo: React.ComponentType<Props> = compose(
-  connect(mapStateToProps),
   translate(),
+  connect(mapStateToProps),
```
Comment (willdurand) [inline src/amo/components/AddonMoreInfo/index.js]: Why is this needed?

Comment (voronianski) [reply src/amo/components/AddonMoreInfo/index.js]: I changed the order because `getVersionInfo` inside `mapStateToProps` needs `i18n` prop.

Comment (willdurand) [reply src/amo/components/AddonMoreInfo/index.js]: ah, I see. Thanks!

**Comment by willdurand** [comment]
Thank you for your first contribution @voronianski 🎉 /cc @caitmuenster

**Comment by caitmuenster** [comment]
Woohoo! Thanks for the patch, @voronianski! ✨ Your contribution has been added to our [recognition wiki](https://wiki.mozilla.org/Add-ons/Contribute/Recognition) and your mozillians profile has been vouched. 

Welcome onboard! We look forward to seeing you around the project.

**Comment by voronianski** [comment]
> Woohoo! Thanks for the patch, @voronianski! ✨ Your contribution has been added to our [recognition wiki](https://wiki.mozilla.org/Add-ons/Contribute/Recognition) and your mozillians profile has been vouched.
> 
> Welcome onboard! We look forward to seeing you around the project.

Thanks!
